### PR TITLE
Allow generic search result items for search providers

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Description
Just tried writing a basic search provider while working on #10067 and realized search results were hardcoded to be webdav-specific. Figured I'd allow for a more generic approach, at least from a type perspective 🙃

### CODE REVIEW

File: packages/web-pkg/src/composables/piniaStores/extensionRegistry.ts [hunk 1]
```diff
@@ -7,11 +7,12 @@ import { Item } from '@ownclouders/web-client/src/helpers'
 import { useConfigStore } from './config'
 import { FolderView } from '../../ui'
 
+export type AvailableExtensions = 'action' | 'search' | 'sidebarNav' | 'sidebarPanel' | 'folderView'
```
Comment (pascalwengerter) [inline packages/web-pkg/src/composables/piniaStores/extensionRegistry.ts:10]: @dschmidt are you fine with this?

Comment (dschmidt) [reply packages/web-pkg/src/composables/piniaStores/extensionRegistry.ts:10]: If you only allow these as type... you are preventing apps from registering their own extension types 😏

Comment (pascalwengerter) [reply packages/web-pkg/src/composables/piniaStores/extensionRegistry.ts]: Good point, thx!

File: packages/web-pkg/src/components/Search/types.ts [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+  name: string
+  icon?: string
+}
 
 export interface SearchResultValue {
-  id: string | number
-  data: SearchResource
+  id: string
+  // FIXME: SearchResource is very specific for webdav search
+  data: GenericSearchResultItem | SearchResource
```
Comment (JammingBen) [inline packages/web-pkg/src/components/Search/types.ts:13]: Just thinking, do we really need the `SearchResource` here then? 🤔 Shouldn't it be as generic as possible in `web-pkg`?

Search implementations like the webdav search could still have their own type, as long as they implement the generic type.

Comment (pascalwengerter) [reply packages/web-pkg/src/components/Search/types.ts:13]: The purpose of this PR was to widen the current typing a little, I don't have a strong opinion on how generic it ideally _should_ be 😇

Comment (JammingBen) [reply packages/web-pkg/src/components/Search/types.ts:13]: Alright, I guess the purpose of this PR is served with this 😄

**Review by JammingBen — APPROVED** [review]

**Review by kulmann — APPROVED** [review]

**Pull request merged**